frontend/src/components/Logo.tsx

```diff
@@ -7,6 +7,22 @@ function Logo() {
   const { t } = useTranslation();
   const { settings, loadingSettings } = useSettings();
   const { logo, loadingFile } = useLogo(settings.customLogoS3Key);
+  const [toHide, setToHide] = useState<boolean>(true);
+
+  // firstUpdate stops useEffect from executing after the first render.
+  const firstUpdate = useRef(true);
+  const secondUpdate = useRef(true);
+  useEffect(() => {
+    if (secondUpdate.current) {
+      if (firstUpdate.current) {
+        firstUpdate.current = false;
+        return;
+      }
+      secondUpdate.current = false;
+      return;
+    }
+    setToHide(false);
+  }, [loadingFile]);
```

Comment:
If we change `loadingFile` to `logo`, we can prevent useEffect from running two times, and it will only run once. In this case, we can remove the complexity of having the secondUpdate useRef.

Reply:
useEffect cannot depend on logo because if the logo never loads (in other words we are using the default logo), we will never set toHide to false. That means the default logo will never show up. I do agree the logic here is not straight-forward enough.